torchgeo/datasets/geo.py

```diff
@@ -349,7 +349,8 @@ def __init__(
                     with rasterio.open(filepath) as src:
                         # See if file has a color map
                         try:
-                            self.cmap = src.colormap(1)
+                            if len(self.cmap) == 0:
```

Comment:
Would there be any advantage of moving this if-statement outside of the try-except block? Trying to think if it would be any faster for datasets with thousands of files...

Reply:
my gut feeling is that the file IO checking whether src has a .colormap will dominate

Reply:
its a bit faster

![image](https://user-images.githubusercontent.com/1088078/154871587-3fca2188-0285-4618-9e3f-acb2066fe5d1.png)

Reply:
That's not really a direct comparison because one has an if statement and one doesn't. The difference is whether or not it hits the try-except statement.

Reply:
What? Doing the file I/O and nothing else takes 1.91, the additional try/except takes .2 no matter what

Reply:
![image](https://user-images.githubusercontent.com/1088078/154880770-569d3bfa-fa52-43d7-b0aa-ef0bbffc0a15.png)